src/serialize.jl

```diff
@@ -0,0 +1,18 @@
+const _pickle = PyNULL()
+
+pickle() = _pickle.o == C_NULL ? copy!(_pickle, pyimport(PyCall.pyversion.major ≥ 3 ? "pickle" : "cPickle")) : _pickle
+
+function Base.serialize(s::AbstractSerializer, pyo::PyObject)
+    pyo.o == C_NULL && return invoke(serialize, (AbstractSerializer, Any), s, pyo)
+    Base.serialize_type(s, PyObject)
+    b = PyBuffer(pycall(pickle()["dumps"], PyObject, pyo))
```

Comment:
You're right; if I handle both input and output PyNULL PyPtr serialization under PyObject it'll be fine.